### ISSUE TITLE
tracing and cacheControl doesn't work with custom field resolver

### ISSUE THREAD

**Issue by chentsulin** [apollographql/apollo-server#865]
Hi, I found that setting `tracing` and `cacheControl` to true breaks apps which support custom `fieldResolver`. For example, I add those lines of code to support resolving fields with `Bookshelf` ORM model: 

```js
// https://github.com/graphql/graphql-js/blob/8913bf0e91789752d2a65179a53de40067a6a41d/src/execution/execute.js#L1308-L1322
function fieldResolver(source, args, context, info) {
  // ensure source is a value for which property access is acceptable.
  if (typeof source === 'object' || typeof source === 'function') {
    const property = source[info.fieldName];
    if (typeof property === 'function') {
      return source[info.fieldName](args, context, info);
    }

    // Important: support field getter for Bookshelf models
    if (typeof source.get === 'function') {
      return source.get(info.fieldName);
    }

    return property;
  }
}

const handleGraphQL = graphqlKoa(ctx => ({
  schema,
  context: ctx,
  fieldResolver,
  debug: process.env.NODE_ENV === 'development',
  tracing: true,
  cacheControl: true,
}));
```

After enabling `tracing` and `cacheControl`, I get `Error: Cannot return null for non-nullable field` error. And if I do `JSON.stringify` manually to my Bookshelf model before passing it as resolver's result, it works.

```js
{
  Query: {
    project: async (obj, args) => {
      const project = await Project.findById(args.id);

      return JSON.stringify(project);
    }
  }
}
```

**Comment by vieiralucas** [comment]
Looks like this is duplication of #716

**Comment by chentsulin** [comment]
We should allow custom field resolvers here to fix it:

https://github.com/apollographql/graphql-extensions/blob/674e8fc51b32cfd9937ed94f495d77b006d536fa/src/index.ts#L153-L155

**Comment by chentsulin** [comment]
PR in graphql-extensions:
https://github.com/apollographql/graphql-extensions/pull/11

**Comment by chentsulin** [comment]
@stubailo Any idea on this? I can handle the PR when I know how to change APIs to pass `fieldResolver` option.

**Comment by glasser** [comment]
This will be fixed in apollo-server 2.0; it will be fixed on the version-2 branch as soon as https://github.com/apollographql/apollo-server/pull/1128 merges (which upgrades graphql-extensions to beta 10, which has a version of @chentsulin's patch, which was actually required to make the tests pass!)

**Comment by rlech** [comment]
I have the same problem using a custom fieldResolver when I use `mocks` or `mockEntireSchema: false`.
